devel/python/python/ert/enkf/export/gen_data_observation_collector.py

```diff
@@ -33,44 +33,39 @@ def getObservationKeyForDataKey(ert, data_key, data_report_step):
         return observation_key
 
 
-
     @staticmethod
-    def loadGenDataObservations(ert, case_name, keys=None):
+    def loadGenDataObservations(ert, case_name, key=None):
         """
         @type ert: EnKFMain
         @type case_name: str
-        @type key: list of str
+        @type key: name of an observation key
         @rtype: DataFrame
         """
         fs = ert.getEnkfFsManager().getFileSystem(case_name)
 
-        observation_keys = GenDataObservationCollector.getAllObservationKeys(ert)
-        if keys is not None:
-            observation_keys = [key for key in keys if key in observation_keys] # ignore keys that doesn't exist
+        available_observation_keys = GenDataObservationCollector.getAllObservationKeys(ert)
+        if key is None or not key in available_observation_keys:
+            raise KeyError("Key '%s' is not a valid observation key")
 
-        columns = observation_keys
-        std_columns = ["STD_%s" % key for key in observation_keys]
+        columns = [key]
+        std_columns = ["STD_%s" % key]
 
         enkf_obs = ert.getObservations()
 
-        max_size = 0
-        for key in observation_keys:
-            obs_vector = enkf_obs[key]
-            report_step = obs_vector.activeStep()
-            obs_node = obs_vector.getNode(report_step)
-            """ :type: ert.enkf.observations.GenObservation """
-            max_size = max(obs_node.getSize(), max_size)
-
-        data = DataFrame(index=range(max_size), columns=columns + std_columns)
-        for key in observation_keys:
-            obs_vector = enkf_obs[key]
-            report_step = obs_vector.activeStep()
-            obs_node = obs_vector.getNode(report_step)
-            """ :type: ert.enkf.observations.GenObservation """
-
-            for iobs , (value,std) in enumerate(obs_node):
-                data_index = obs_node.getDataIndex( iobs )
-                data[key][data_index] = value
-                data["STD_%s" % key][data_index] = std
+        index_set = set()
+        obs_vector = enkf_obs[key]
+        report_step = obs_vector.activeStep()
+        obs_node = obs_vector.getNode(report_step)
+        """ :type: ert.enkf.observations.GenObservation """
+        for obs_index in range(len(obs_node)):
+            index_set.add(obs_node.getIndex(obs_index))
+
+        index_list = sorted(list(index_set))
+        data = DataFrame(index=index_list, columns=columns + std_columns)
+
+        for obs_index, (value, std) in enumerate(obs_node):
+            data_index = obs_node.getIndex(obs_index)
```

Comment:
That is not what I want. I want the observation value and its index in "GenData space" so that values are properly aligned when plotting along with normal GenData.